solution_runner/commands/commands_utils.py

```diff
@@ -0,0 +1,89 @@
+import urllib.parse
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
+from typing import Any
+
+import click
+import requests
+import yaml
+
+from . import consts
+
+
+class PathType(Enum):
+    FILE = 1
+    DIRECTORY = 2
+
+
+def get_setting(key: str) -> Any:
+    """
+    :param key: key to retrieve its value
+    :return: value of `key` which is stored in the configuration file
+    """
+    configuration_file = consts.APP_DATA_DIRECTORY / consts.CONFIGURATION_FILE_NAME
+    try:
+        configuration = yaml.safe_load(configuration_file.read_text())
```

Comment:
If you are getting more than one settings per program execution, it might be better to load the configuration only once and cache the dict in memory.

Reply:
> cache the dict in memory

How?

Reply:
Global variable, or defining a function performing `yaml.safe_load(configuration_file.read_text())` and decorating it with `@functools.cache`.